Sources/Paywalls/Components/Common/Dimension.swift

```diff
@@ -0,0 +1,89 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  Dimension.swift
+//
+//  Created by Josh Holtz on 9/27/24.
+// swiftlint:disable missing_docs
+
+import Foundation
+
+#if PAYWALL_COMPONENTS
+
+public extension PaywallComponent {
+
+    enum Dimension: Codable, Sendable, Hashable {
+
+        case vertical(HorizontalAlignment)
+        case horizontal(VerticalAlignment)
+        case zlayer(TwoDimensionAlignment)
```

Comment:
This could also be something like `depth`? Since this is public, feels slightly inconsistent to use a `z` here, but not use an `x` and `y` in the other cases.

Reply:
Ahhhh yes! I was really not liking the name of that one 😅 Depth could totally work 💪